ImageGalery/GalleryCollectionViewController.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  GalleryCollectionViewController.swift
+//  ImageGalery
+//
+//  Created by apple on 18/07/2018.
+//  Copyright © 2018 wolfpack. All rights reserved.
+//
+
+import UIKit
+
+private let reuseIdentifier = "GalleryCollectionViewCell"
+
+
+protocol galleryUpdateDelegate {
+    func UppdateGalery(_ updatedGallery: ImageGallery)
+}
+
+class GalleryCollectionViewController: UICollectionViewController, UICollectionViewDropDelegate, UICollectionViewDelegateFlowLayout {
+    
+    var tableViewController: GaleryTableViewController?
+    var delegate: galleryUpdateDelegate?
+    var gallery: ImageGallery? = ImageGallery(images: [], title: "") {
+        didSet {
+            if gallery!.images.count > 0 {
```

Comment:
this line will crash the app when you set 'gallery' to nil, avoid force unwrapping optionals. Either user a guard let to extract it or don't make it optional (remove the '?'  from 'var gallery: ImageGallery?')